internal/connector/test/integration/feature_test.go

```diff
@@ -82,9 +83,10 @@ func TestMain(m *testing.M) {
 		},
 		connector.ConfigProviders(false),
 	)
+	defer func() { os.Exit(exitCode) }()
 	defer teardown()
```

Comment:
This deferred method was not being called following `os.Exit(..)`.

If you run `make db/login` after running the integration tests you can see the database tables and content are still there... they should have been removed by this `teardown()` method invocation.

Furthermore, you can also verify this by placing the following in `internal/connector/internal/migrations/migrations.go`
```
...
addOrgIDAnnotations("202212050000"),
	&gormigrate.Migration{
		ID: "test",
		Migrate: func(g *gorm.DB) error {
			fmt.Println("============== RUNNING MIGRATION")
			return nil
		},
		Rollback: func(g *gorm.DB) error {
			fmt.Println("============== RUNNING ROLLBACK")
			return nil
		},
...
```
You can observe that the rollback method is not called.

Reply:
I followed the approach suggested [here](https://www.golinuxcloud.com/golang-handle-defer-calls-os-exit/) ("Method-2")

Reply:
Good catch :+1: .

The reason about why deferred was not being called is documented in the `os.Exit` function documentation available in https://pkg.go.dev/os#Exit:

Exit causes the current program to exit with the given status code. Conventionally, code zero indicates success, non-zero an error. The program terminates immediately; **deferred functions are not run**. 

The approach used looks good to me :+1: